tpdatasrc/tpgamefiles/scr/tpModifiers/sp_vigor_fast_healing.py

```diff
@@ -6,21 +6,42 @@
 print "Registering sp-Vigor Fast Healing"
 
 def healTick(attachee, args, evt_obj):
-    duration = args.get_arg
+    healAmount = args.get_arg(2)
     spellId = args.get_arg(0)
     spellEnum = spellEnum = tpdp.SpellPacket(spellId).spell_enum
     spellName = game.get_spell_mesline(spellEnum)
     spellTag = "TAG_SPELLS_{}".format(spellName.upper().replace(" ", "_"))
-    healAmount = args.get_arg(2)
     ### workaround for heal ###
     #heal requires a dice
     healDice = dice_new('1d1')
     healDice.bonus = healAmount -1
     ### workaround end ###
     game.particles ('sp-Vigor', attachee)
-    attachee.heal(attachee, healDice, D20A_HEAL, 0)
-    attachee.healsubdual(attachee, healDice, D20A_HEAL, 0)
-    game.create_history_freeform("{} is healed for {} by ~{}~[{}]\n\n".format(attachee.description, healAmount, spellName, spellTag))
+    if attachee.obj_get_int(obj_f_critter_subdual_damage):
+        attachee.healsubdual(attachee, healDice, D20A_HEAL, 0)
+    elif attachee.obj_get_int(obj_f_hp_damage):
+        attachee.heal(attachee, healDice, D20A_HEAL, 0)
+    if game.combat_is_active(): #Limit History output to combat only
+        game.create_history_freeform("{} is healed for {} by ~{}~[{}]\n\n".format(attachee.description, healAmount, spellName, spellTag))
+    return 0
+
+#Adding an onRemoveHealTick check for triggering a Rest/Passing Time
+#While a vigor effect is ticking; This will sum up leftover ticks in one heal
```

Comment:
IIRC the onBeginRound event also tells you how much time has passed so you shouldn't need this workaround

Reply:
> IIRC the onBeginRound event also tells you how much time has passed so you shouldn't need this workaround

Here is what I tried:

- My first idea was to simply to add a hook for ET_OnNewDay. To me, this seemed to be the hook use, as it avoids any other reason the spell condition is removed (like getting killed or the effect getting dispelled) but it didn't work as the spell condition gets removed before the hook actually gets triggered by the New Day Event.
- Then I tried to modifiy the already existing ET_OnBeginRound hook by simply adding an extra check (if evt_obj.data1 != 1) but to my surprise the OnBeginRound Hook is not triggered when you hit the rest button.
- Then I tried to use the ET_OnConditionRemove Hook, but this adds way to much headache, as it is triggered by too many events (e.g. TeleportReconnect, Dispel, etc) so I would need alot of case handling just to handle an edge case. This didn't seem worth it.

So atm I did scrap the heal sum idea, I admit I still would like to have it, but I am out of ideas how to do it properly.